src/glob.zig

```diff
@@ -361,8 +367,54 @@ pub fn GlobWalker_(
 
             pub fn init(this: *Iterator) !Maybe(void) {
                 log("Iterator init pattern={s}", .{this.walker.pattern});
+                var was_absolute = false;
+                const root_work_item = brk: {
+                    var use_posix = bun.Environment.isPosix;
+                    const is_absolute = if (bun.Environment.isPosix) std.fs.path.isAbsolute(this.walker.pattern) else std.fs.path.isAbsolute(this.walker.pattern) or is_absolute: {
+                        use_posix = true;
+                        break :is_absolute std.fs.path.isAbsolutePosix(this.walker.pattern);
+                    };
+
+                    if (!is_absolute) break :brk WorkItem.new(this.walker.cwd, 0, .directory);
+
+                    was_absolute = true;
+
+                    const path_without_special_syntax = this.walker.pattern[0..this.walker.end_byte_of_basename_excluding_special_syntax];
+                    const component_idx = this.walker.basename_excluding_special_syntax_component_idx + 1;
+
+                    // This means we got a pattern without any special glob syntax, for example:
+                    // `/Users/zackradisic/foo/bar`
+                    // In that case we don't need to do any walking and can just open up the FS entry
+                    if (component_idx >= this.walker.patternComponents.items.len) {
+                        const path = try this.walker.arena.allocator().dupeZ(u8, path_without_special_syntax);
+                        const fd = switch (try Accessor.open(path)) {
+                            .err => |e| {
+                                if (e.getErrno() == bun.C.E.NOTDIR) {
+                                    // TODO check symlink
+                                    this.iter_state = .{ .root_matched = path };
+                                    return Maybe(void).success;
+                                }
+                                const errpath = try this.walker.arena.allocator().dupeZ(u8, path);
+                                return .{ .err = e.withPath(errpath) };
```

Comment:
I suspect there will be more errors to ignore here. Like EPERM, EACCESS, EBUSY etc